packages/network/test/integration/tracker-node-disconnects-on-invalid-status.test.ts

```diff
@@ -0,0 +1,54 @@
+import { runAndWaitForEvents } from "streamr-test-utils"
+import { Tracker} from "../../src/logic/tracker/Tracker"
+import { NodeToTracker, Event as NodeToTrackerEvent } from "../../src/protocol/NodeToTracker"
+import { PeerInfo } from "../../src/connection/PeerInfo"
+import NodeClientWsEndpoint from "../../src/connection/ws/NodeClientWsEndpoint"
+import { startTracker } from "../../src/composition"
+import { Status } from "../../src/identifiers"
+
+describe('Tracker disconnects from node if node sends invalid status data', () => {
+    let tracker: Tracker
+    let nodeToTracker: NodeToTracker
+    const TRACKER_ID = 'tracker'
+
+    beforeEach(async () => {
+        tracker = await startTracker({
+            listen: {
+                hostname: '127.0.0.1',
+                port: 30436
+            },
+            id: 'tracker'
+        })
+        const peerInfo1 = PeerInfo.newNode('nodeToTracker1')
+        const trackerPeerInfo = PeerInfo.newTracker(TRACKER_ID)
+        const wsClient = new NodeClientWsEndpoint(peerInfo1)
+        nodeToTracker = new NodeToTracker(wsClient)
+
+        await runAndWaitForEvents([
+            () => nodeToTracker.connectToTracker(tracker.getUrl(), trackerPeerInfo)
+        ], [
+            [nodeToTracker, NodeToTrackerEvent.CONNECTED_TO_TRACKER]
+        ])
+    })
+    beforeEach(async () => {
+        await tracker.stop()
+        await nodeToTracker.stop()
```

Comment:
- `allSettled` to ensure that both `stop` methods get invoked (even if `tracker.stop()` rejected)
-  `?.` in case `beforeEach` threw exception in the middle (i.e. `tracker` or `nodeToTracker` was never initialized)
```suggestion
        Promise.allSettled([
            await tracker?.stop()
            await nodeToTracker?.stop()
        ])
```